### PR TITLE
Add sys_mdbg_service case 01

### PR DESCRIPTION
**Pull request by VocalFan**
This PR adds sys_mdbg_service or, specifically, case 01 which requests for a namedobj to be created using args 1 onwards. To make this work, this also refactors the way namedobj_create is called.

### CODE REVIEW

**Comment by VocalFan** [comment]
@ultimaweapon May I request judgement here? XD

File: src/kernel/src/namedobj/mod.rs [hunk 1]
```diff
@@ -11,17 +11,20 @@ impl NamedObjManager {
     pub fn new(sys: &mut Syscalls) -> Arc<Self> {
         let namedobj = Arc::new(Self {});
 
-        sys.register(557, &namedobj, Self::sys_namedobj_create);
+        sys.register(557, &Arc::clone(&namedobj), Self::sys_namedobj_create);
```
Comment (SuchAFuriousDeath) [inline src/kernel/src/namedobj/mod.rs:14]: tf is this? :eyes:

**Review by SuchAFuriousDeath — CHANGES_REQUESTED** [review]

File: src/kernel/src/namedobj/mod.rs (hunk 1 above)
Comment (VocalFan) [reply src/kernel/src/namedobj/mod.rs:14]: Without cloning, it sort of panics in the register-

Comment (VocalFan) [reply src/kernel/src/namedobj/mod.rs:14]: @SuchAFuriousDeath If I can find a way to make namedobj_create a public function, then I'll move mdbg.

Comment (SuchAFuriousDeath) [reply src/kernel/src/namedobj/mod.rs:14]: > Without cloning, it sort of panics in the register-

I don't believe that. Everywhere else it's fine.

Comment (VocalFan) [reply src/kernel/src/namedobj/mod.rs:14]: Nevermind, I think I did that as a pseudo fix to an earlier bug... Seems to work fine without cloning.

**Review by mikusp — APPROVED** [review]

File: src/kernel/src/namedobj/mod.rs [hunk 2]
```diff
@@ -42,6 +45,38 @@ impl NamedObjManager {
...
+
+        self.namedobj_create(td, name, data, flags)
+    }
+
+    fn sys_mdbg_service(self: &Arc<Self>, td: &VThread, i: &SysIn) -> Result<SysOut, SysErr> {
+        // Get the mdbg arg
+        let mdbg_type: i32 = i.args[0].try_into().unwrap();
+
+        match mdbg_type {
+            0x1 => {
```
Comment (SuchAFuriousDeath) [inline src/kernel/src/namedobj/mod.rs:67]: If you look into ghidra, you'll see that this is wrong.
This whole chucnk is missing.
image

**Review by SuchAFuriousDeath — CHANGES_REQUESTED** [review]

**Pull request closed without merging**